packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/petrinet/SpeciesJsonDeserializer.java

```diff
@@ -0,0 +1,46 @@
+package software.uncharted.terarium.hmiserver.models.petrinet;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class SpeciesJsonDeserializer extends JsonDeserializer<Species> {
+	/*
+		{
+			"sname": "S",
+			"mira_ids": "[('identity', 'ido:0000514'), ('identity', 'ido:0000511')]",
+			"mira_context": [('identity', 'ido:0000514')]",
+		}
+	 */
+
+	@Override
+	public Species deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException, JacksonException {
+		JsonNode node = p.getCodec().readTree(p);
+
+		final Species species = new Species();
+		species.setName(node.get("sname").asText());
+		species.setMiraIds(new ArrayList<>());
+		species.setMiraContext(new ArrayList<>());
```

Comment:
This is totally correct and doesn't have to change, but as someone who has been recently introduced to lombok, if you add the annotation `@Accessors(chain = true)` to your `Species` class you would be able to chain these together like...
`final Species species = new Species().setName(foo).setMiraIds(bar).setMiraContext(fooBar);`

Again... totally not needed but kind of cool!

Reply:
@dvince2 There is also `@builder` ... if we want to fully lombok there is also `val` keyword 🤔 ... any thoughts on where we draw the line?